mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -3271,11 +3298,17 @@ LogicalResult LLVMDialect::verifyRegionResultAttribute(Operation *op,
 
 Operation *LLVMDialect::materializeConstant(OpBuilder &builder, Attribute value,
                                             Type type, Location loc) {
-  // If this was folded from an llvm.mlir.addressof operation, it should be
-  // materialized as such.
+  // If this was folded from a ConstantLike operation, it should be materialized
```

Comment:
```suggestion
  // If this was folded from another operation than llvm.mlir.constant op, it should be materialized
```
nit: the constant op is also ConstantLike. I think I would explicitly say not mlir.constant somehow.

Reply:
I've also added a comment on ZeroOp specifically which may be folded into the built-in zero attribute.